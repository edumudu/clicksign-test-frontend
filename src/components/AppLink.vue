<template>
  <a
    v-if="isExternal"
    target="_blank"
    rel="noopener noreferrer"
    :href="to"
  >
    <slot />
  </a>

  <RouterLink
    v-else
    v-bind="$props"
  >
    <slot />
  </RouterLink>
</template>

<script lang="ts">
import { defineComponent, computed } from 'vue';
import { RouterLink } from 'vue-router';

export default defineComponent({
  name: 'AppLink',

  props: {
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    /// @ts-ignore
    ...RouterLink.props,
  },

  setup(props) {
    const isExternal = computed(() => props.to && typeof props.to === 'string' && props.to.startsWith('http'));

    return {
      isExternal,
    };
  },
});
</script>
