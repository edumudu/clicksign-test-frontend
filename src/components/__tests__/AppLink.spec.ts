import { shallowMount, VueWrapper, RouterLinkStub } from '@vue/test-utils';

import AppLink from '../AppLink.vue';

describe('AppLink.vue', () => {
  let wrapper: VueWrapper<any>;

  beforeEach(() => {
    wrapper = shallowMount(AppLink, {
      props: {
        to: '/',
      },

      global: {
        stubs: {
          RouterLink: RouterLinkStub,
        },
      },
    });
  });

  it('should match snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('Should render a external link when link starts with http', async () => {
    await wrapper.setProps({ to: 'http://google.com' });

    expect(wrapper.attributes('href')).toBe('http://google.com');
  });

  it('Should render RouterLink link when link not starts with http', async () => {
    await wrapper.setProps({ to: '/path' });

    expect(wrapper.findComponent(RouterLinkStub).exists()).toBeTruthy();
  });

  it('Should add target="_blank" and security attributes to external link', async () => {
    await wrapper.setProps({ to: 'http://google.com' });

    expect(wrapper.attributes('target')).toMatch('_blank');
    expect(wrapper.attributes('rel')).toMatch('noopener noreferrer');
  });

  it('Should allow render a target="_self" external link', async () => {
    await wrapper.setProps({ to: 'http://google.com', target: '_self' });

    expect(wrapper.attributes('target')).toMatch('_self');
  });
});
