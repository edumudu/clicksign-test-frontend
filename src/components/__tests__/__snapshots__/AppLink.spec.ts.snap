// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLink.vue should match snapshot 1`] = `
<a
  ariacurrentvalue="page"
  custom="false"
  replace="false"
>
  
  
</a>
`;
